items.json

```
@@ -4,45 +4,121 @@
     "name": "Fabric",
     "description": "A common crafting material.",
     "type": "Material",
-    "imageFilename": "https://cdn.arctracker.io/items/fabric.png"
+    "imageFilename": "https://cdn.arctracker.io/items/fabric.png",
+    "craftable": false,
+    "craftableUsing": [],
+    "recyclesInto": [],
+    "usedForCrafting": [
+      "Sterilized Bandage",
+      "Rope",
+      "Herbal Bandage",
+      "Equipment Bench Level I",
+      "Med Station Level I",
+      "Polluted Air Filter"
+    ]
```

Would it be better to have these linked via ID instead arbitrary strings (if available)?

```suggestion
    "usedForCrafting": [
      "sterilized_bandage",
      "rope",
      "herbal_bandage",
      "Equipment Bench Level I",
      "Med Station Level I",
      "polluted_air_filter"
    ]
```

Not sure how to handle non-item items like Bench Upgrades.

Maybe do a lookup on the ID and if no item is matched just display the string as-is?

---

Or maybe a similar structure to `craftableUsing`.

---

Good call on the itemIds. For the hideouts lets use the id in the hideoutsModules.json. and maybe just a object for the level(s) its used in.

---

@blzaugg good call, it should definitely use the ids! I was transferring over from a set of data where `name` = `id` and missed converting this list to use `id`s

@mattmarcin another good point - I haven't brought in the objects from `hideoutModules.json` yet but can pull from there as well. Im using DynamoDB on my project and I think it will complain about mixed types like 

```
"usedForCrafting": [
      "sterilized_bandage",
      {
          "id": "weapon_bench",
          "level": 1,
    }
]
```

but I think I can (and probably should) just treat this as a list of objs like `recyclesInto` and `craftableUsing`

```
"usedForCrafting": [
       {
           "id": "sterilized_bandage",
           "name": "Sterilized Bandage"
       },
      {
          "id": "weapon_bench",
          "name": "Weapon Bench",
          "level": 1,
    }
]
```

---

You can probably omit name for the hideout since it's in the hideout json. 

Also just remembered. I believe there is an example in the items json of items with a crafting recipe. This would probably simplify things. Thoughts?

---

I should've put this in the original PR too my bad - I did see you had `crafting` information available like this for `raider_augment_iv` (as an example)

```
"crafting": {
      "power_rod": 1,
      "processor": 4
    },
```

I changed it to this

```
"craftableUsing": [
  {
    "amount": 1,
    "itemName": "Power Rod",
    "itemId": "power_rod"
  },
  {
    "amount": 4,
    "itemName": "Processor",
    "itemId": "processor"
  }
],
```

I did this with the API in mind, thinking that the list structure would lead to less API calls. As in if an item was selected in the UI, I could display its crafting materials with the normal name `Power Rod` instead of just having `power_rod` and needing another API call to get `name`. 

Realizing that this might make less sense for just doing the lookups in JSON 🤔 

What if we included both? So the definition of `raider_augment_iv` now looks like 
```
{
    "id": "raider_augment_iv",
    "name": "Raider Augment IV",
    "description": "Can be equipped to gain more loadout space and carry capacity...",
    "craftableUsing": [
        {
            "itemId": "power_rod",
            "amount": 1,
            "itemName": "Power Rod"
        },
        {
            "itemId": "processor",
            "amount": 4,
            "itemName": "Processor"
        }
    ],
    "crafting": {
      "power_rod": 1,
      "processor": 4
    },
    .......
}
```
A bit redundant but avoids any breaking change's currently using `crafting`

---

I think we should stick with the crafting using the itemIds so prevent duplication. Anyone using this JSON should load all the JSON data at once so there wouldn't be multiple API calls.

The issue with the craftableUsing method is if a name changes it needs to be updated in multiple places. It also makes localization more difficult if we end up adding translations later.

---

Taking a smaller swing at this with the more recent commit, let me know your thoughts @mattmarcin 

I can clean up these empty properties too just lmk (might make the review more manageable) 
https://github.com/RaidTheory/arcraiders-data/pull/3/files#diff-d907dcbebfe7d89f411d1b1883be64009f83d87ab7bdd634fe6b91818b658f7eR18-R20

---

Edited in latest commit